mllib/src/main/scala/org/apache/spark/ml/optim/NormalEquationSolver.scala

```diff
@@ -109,6 +109,11 @@ private[optim] class QuasiNewtonSolver(
       state = states.next()
       arrayBuilder += state.adjustedValue
     }
+    if (state == null || state.searchFailed) {
+      val msg = s"${optimizer.getClass.getName} failed."
+      logError(msg)
```

Comment:
Warning actually seems more appropriate in all of these cases. It's not an unrecoverable Spark error.

Reply:
Thanks for your review.
Is this a comment about the logging level?

Reply:
Yeah, I'm thinking logWarning is more appropriate. I know the existing code had logError, and could change too.

Reply:
I got it. Fixed.